jit/src/main/java/com/justintime/jit/service/impl/DashboardServiceImpl.java

```diff
@@ -41,20 +42,32 @@ public class DashboardServiceImpl implements DashboardService {
 
     @Autowired
     private UserRepository userRepository;
+    
+    @Autowired
+    private RestaurantRoleRepository restaurantRoleRepository;
 
 
     public DashboardDTO getDashboardData(String restaurantCode) {
         Restaurant restaurant = restaurantRepository.findByRestaurantCode(restaurantCode).orElseThrow(() -> new ResourceNotFoundException("Restaurant not found with restaurant code" + restaurantCode));
         DashboardDTO dashboardDTO = new DashboardDTO();
+        
+        // Get role IDs for this restaurant by roleType
+        RestaurantRole cookRole = restaurantRoleRepository.findFirstByRoleTypeAndRestaurantCode(Role.COOK, restaurantCode)
+                .orElse(null);
+        RestaurantRole serverRole = restaurantRoleRepository.findFirstByRoleTypeAndRestaurantCode(Role.SERVER, restaurantCode)
+                .orElse(null);
+        RestaurantRole customerRole = restaurantRoleRepository.findFirstByRoleTypeAndRestaurantCode(Role.CUSTOMER, restaurantCode)
+                .orElse(null);
+        
```

Comment:
Its better to have a null check 
```suggestion
        Long restaurantId = restaurant != null ? restaurant.getId() : null;

if (restaurantId != null) {
    dashboardDTO.setTotalCooks(
        cookRole != null ? userRepository.countByRestaurantIdAndRoleId(restaurantId, cookRole.getId()) : 0
    );
    dashboardDTO.setTotalServers(
        serverRole != null ? userRepository.countByRestaurantIdAndRoleId(restaurantId, serverRole.getId()) : 0
    );
} else {
    dashboardDTO.setTotalCooks(0);
      ......;
}
```